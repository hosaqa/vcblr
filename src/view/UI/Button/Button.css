.Button {
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: 0;
  padding: 0;
  background-color: transparent;
  box-shadow: none;
  cursor: pointer;
  user-select: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  touch-action: manipulation;
  display: inline-flex;
  align-items: center;
  justify-content: center;
}

.Button--primary {
  color: var(--font-secondary);
  background-color: var(--accent);
  border-radius: var(--border-radius);
  padding: 7px 15px;
}

.Button:disabled,
.Button[disabled='disabled'] {
  pointer-events: none;
  background-color: var(--disabled);
}
