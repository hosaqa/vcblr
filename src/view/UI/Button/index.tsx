import Buttom from './Button';

export default Buttom;
