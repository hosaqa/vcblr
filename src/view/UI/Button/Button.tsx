import React from 'react';
import clsx from 'clsx';
import './Button.css';

interface IButton extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: string;
  className?: string;
  children?: React.ReactNode;
}

const Button: React.FC<IButton> = ({
  variant = 'primary',
  className,
  children,
  ...props
}) => (
  <button
    className={clsx('Button', `Button--${variant}`, className)}
    {...props}
  >
    {children}
  </button>
);

export default Button;
