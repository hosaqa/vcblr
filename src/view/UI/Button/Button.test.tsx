import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button UI', () => {
  const props = {
    children: 'button text',
  };

  describe('default params and text', () => {
    const button = shallow(<Button {...props} />);

    it('renders properly', () => {
      expect(button).toMatchSnapshot();
    });

    it('text renders properly', () => {
      expect(button.find('button').text()).toEqual(props.children);
    });

    it('properly classnames', () => {
      expect(button.find('button').hasClass('Button')).toEqual(true);
      expect(button.find('button').hasClass('Button--primary')).toEqual(true);
    });
  });

  describe('secondary type and custom classname', () => {
    const button = shallow(
      <Button className="custom" variant="secondary" {...props} />
    );

    it('renders properly', () => {
      expect(button).toMatchSnapshot();
    });

    it('text renders properly', () => {
      expect(button.find('button').text()).toEqual(props.children);
    });

    it('properly classnames', () => {
      expect(button.find('button').hasClass('Button')).toEqual(true);
      expect(button.find('button').hasClass('Button--secondary')).toEqual(true);
      expect(button.find('button').hasClass('custom')).toEqual(true);
    });
  });

  describe('html attributes', () => {
    const ariaLabelText = 'Menu button';

    const button = shallow(
      <Button disabled aria-label={ariaLabelText} {...props} />
    );

    it('renders properly', () => {
      expect(button).toMatchSnapshot();
    });

    it('text renders properly', () => {
      expect(button.find('button').text()).toEqual(props.children);
    });

    it('aria-label renders properly', () => {
      expect(button.find('button').prop('aria-label')).toEqual(ariaLabelText);
    });

    it('disabled attribute', () => {
      expect(button.find('button').prop('disabled')).toBeTruthy();
    });
  });
});
