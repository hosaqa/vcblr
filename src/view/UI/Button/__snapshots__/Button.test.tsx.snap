// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button UI default params and text renders properly 1`] = `ShallowWrapper {}`;

exports[`Button UI html attributes renders properly 1`] = `ShallowWrapper {}`;

exports[`Button UI secondary type and custom classname renders properly 1`] = `ShallowWrapper {}`;
